### ISSUE TITLE
Office 2010: Spelling Errors Incorrectly Cached when Reading Current Line and Moving Focus

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#10106]
### Steps to reproduce:

1. Ensure that NVDA is set to announce spelling errors, and open either Outlook or Word 2010

2. Deliberately type a line or word of text that is detected as an error by Office, and ensure that NVDA announces this as an error when placing focus on the word - for example: "Tis is a line of text"

3. Ensuring that focus is at the end of the line, perform a read current line; you'll notice that NVDA announces the fact that the line contains an error

4. Finally, use either the left/right arrow keys to move focus

### Actual behavior:

NVDA will announce that you are out of the error, despite the fact that focus was never on the error in the first place - NVDA has cached this error from when the line was read

### Expected behavior:

NVDA shouldn't announce that focus was moved away from the error if focus was not on the error text at the time that the cursor was moved - as an alternative, we could consider not announcing spelling/grammer errors when reading the current line

#### NVDA installed/portable/running from source:

Installed copy

#### NVDA version:

alpha-18420,a8e8db19

#### Windows version:

Windows 7 X64

#### Name and version of other software in use when reproducing the issue:

Microsoft Office 2010

#### Does the issue still occur after restarting your PC?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No

**Comment by lukaszgo1** [comment]
I cannot reproduce this issue with exactly same setup as your the only exception being the fact, that NVDA is a portable version. Have you tried with add-ons disabled, and if it wouldn't help with a default config?

**Comment by lukaszgo1** [comment]
Ah,, one more think, what synthesizer was used?

**Comment by Adriani90** [comment]
I can reproduce this issue in NVDA 2019.2 with addons disabled and eSpeak also on Office 365.
Repro as follows with keyboard layout laptop:
1. In document formating settings, enable spelling errors to be reported 
2. Open Word
3. Type helo
4. Press nvda+l
5. Press left arrow
Actual: NVDA reports spelling error in step 4 but does not report it in step 5 when focusing the last letter of helo. Instead, NVDA reports "end of spelling error" when pressing right arrow after focusing the letter o.
Expected: NVDA should report spelling error after pressing left arrow in step 5.

6. Type helo 
7. repeat steps 4 and 5.
Actual: when typing a space after helo, something happens. In case of step 5, NVDA reports end of spelling error although I did never focus the word itself. It was only reported when I pressed nvda+l.
Expected: NVDA should report space when pressing left arrow without reporting "end of spelling error".

**Comment by Adriani90** [comment]
This is also reproducible when selecting a word and typing another one, then typing space and then pressing left arrow. So it seems not only linked to the read line feature.

**Comment by XLTechie** [comment]
elliott94 wrote:

> alternative, we could consider not announcing spelling/grammer errors when reading the current line

I strongly oppose that alternative.

**Comment by Brian1Gaff** [comment]
In case its important, there was an update for Office 2010 that came down 
with this months update to Windows.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by DrSooom** [comment]
I don't think that KB4475506, KB4475533 and KB4475573 have something to do with this.
